File: Cite.EvalIt/Query/DataObjectReviewLookup.cs
using Cite.EvalIt.Common;
using Cite.Tools.Data.Query;
using System;
using System.Collections.Generic;
using System.Text;

namespace Cite.EvalIt.Query
{
	public class DataObjectReviewLookup : Lookup
	{
		public List<Guid> Ids { get; set; }
		public List<Guid> ExcludedIds { get; set; }
		public List<Guid> ObjectIds { get; set; }
		public List<Guid> UserIds { get; set; }
		public List<IsActive> IsActive { get; set; }

		//public UserQuery Enrich(QueryFactory factory)
		//{
		//	UserQuery query = factory.Query<UserQuery>();

		//	if (this.Ids != null) query.Ids(this.Ids);
		//	if (this.ExcludedIds != null) query.ExcludedIds(this.ExcludedIds);
		//	if (this.IsActive != null) query.IsActive(this.IsActive);

		//	this.EnrichCommon(query);

		//	return query;
		//}
	}
}

File: Cite.EvalIt.WebApp/src/app/core/query/data-object-review.lookup.ts
import { IsActive } from "@app/core/enum/is-active.enum";
import { ReviewAnonymity } from "@app/core/enum/review-anonymity.enum";
import { Lookup } from "@common/model/lookup";
import { Guid } from "@common/types/guid";

export class DataObjectReviewLookup extends Lookup implements DataObjectReviewFilter {
	ids: Guid[];
	excludedIds: Guid[];
	objectIds: Guid[];
	userIds: Guid[];
	isActive: IsActive[];

	constructor() {
		super();
	}
}

export interface DataObjectReviewFilter {
	ids: Guid[];
	excludedIds: Guid[];
	objectIds: Guid[];
	userIds: Guid[];
	isActive: IsActive[];
}
